src/core/prompts/responses.ts

```diff
@@ -43,6 +44,49 @@ Otherwise, if you have not completed the task and do not need additional informa
 		}
 	},
 
+	mcpToolResult: (
+		result?: McpToolCallResponse,
+		supportsImages: boolean = true,
+	): Array<Anthropic.TextBlockParam | Anthropic.ImageBlockParam> => {
+		return (
+			result?.content
+				.map((item) => {
+					if (item.type === "text") {
+						return item as Anthropic.TextBlockParam
+					}
+					if (item.type === "resource") {
+						const { blob, ...rest } = item.resource
+						return {
+							type: "text",
+							text: JSON.stringify(rest, null, 2),
+						} as Anthropic.TextBlockParam
+					}
+					if (item.type === "image") {
+						if (supportsImages) {
+							return {
+								source: {
+									data: item.data,
+									media_type: item.mimeType,
+									type: "base64",
+								},
+								type: "image",
+							} as Anthropic.ImageBlockParam
+						} else {
+							return {
+								type: "text",
+								text: "[Image placeholder (images not supported)]",
+							} as Anthropic.TextBlockParam
+						}
```

Comment:
The else here is redundant, let's remove this.
```suggestion
						}
						return {
							type: "text",
							text: "[Image placeholder (images not supported)]",
						} as Anthropic.TextBlockParam
```